packages/flutter/test/services/set_method_call_handler_test.dart

```diff
@@ -0,0 +1,28 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/services.dart';
+import 'package:flutter_test/flutter_test.dart';
+
+void main() {
+  // We need a separate test file for this test case (instead of including it
+  // in platform_channel_test.dart) since we rely on the WidgetsFlutterBinding
+  // not being initialized and we call ensureInitialized() in the other test
+  // file.
+  test('throws assertion error if WidgetsFlutterBinding is not yet initialized', () {
+    const MethodChannel methodChannel = MethodChannel('mock');
+
+    // This used to throw _CastError:<Null check operator used on a null value>
+    // before the assertion was added, which we want to avoid in order to hint
+    // callers towards how to fix the error.
+    expect(() => methodChannel.setMethodCallHandler(null), throwsAssertionError);
+  });
+
+  test('does not throw when WidgetsFlutterBinding is already initialized', () {
```

Comment:
@gspencergoog will this ever be randomized to run before the first test? if it does, we should probably create a new test file...

Reply:
It seems to pass in CI right now. I am pretty sure **by default**, they run sequentially. Otherwise, the old `flutter_driver` tests would also not have worked (where you used to put several `test()` cases that depended on the previous one finishing).

Reply:
It's might be passing *today*, since the CI seeds are based on the date, but tomorrow it has a 50/50 chance of running in a different order.

By default (in CI, not locally) they run in a random order based on that day's seed. Today's seed is 20210824.

Reply:
@chunhtai @gspencergoog what do you suggest? Combining them into a single test case? Or creating two files for this?

Reply:
Combine them sounds good